evm/src/executor/fork/cache.rs

```diff
@@ -32,13 +32,16 @@ impl BlockchainDb {
     ///   - the file the `cache_path` points to, does not exist
     ///   - the file contains malformed data, or if it couldn't be read
     ///   - the provided `meta` differs from [BlockchainDbMeta] that's stored on disk
-    pub fn new(meta: BlockchainDbMeta, cache_path: Option<PathBuf>) -> Self {
+    pub fn new(meta: BlockchainDbMeta, cache_path: Option<PathBuf>, skip_check: bool) -> Self {
```

Comment:
this is for the "unrelated" metadata issue you mentioned, right?

I'd rather keep the new function and another like `fn with_cache_check(.., bool)` or similar and then new calls this with false